aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,767 @@
+
+        Parameters
+        ----------
+        lif: LifFile
+
+        Returns
+        -------
+        list[dict]
+            A list of dictionaries containing Dimension / depth. If the shape is consistent across Scenes then
+            the list will have only one Dictionary. If the shape is inconsistent the the list will have a dictionary
+             for each Scene. A consistently shaped file with 3 scenes, 7 time-points
+            and 4 Z slices containing images of (h,w) = (325, 475) would return
+            [
+             {'S': (0, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ].
+            The result for a similarly shaped file but with different number of time-points per scene would yield
+            [
+             {'S': (0, 1), 'T': (0,8), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (1, 2), 'T': (0,6), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)},
+             {'S': (2, 3), 'T': (0,7), 'X': (0, 475), 'Y': (0, 325), 'Z': (0, 4)}
+            ]
+
+        """
+        shape_list = [{'T': (0, img.nt),
+                       'C': (0, img.channels),
+                       'Z': (0, img.nz),
+                       'Y': (0, img.dims[1]),
+                       'X': (0, img.dims[0])}
+                      for idx, img in enumerate(lif.get_iter_image())]
+        consistent = all(elem == shape_list[0] for elem in shape_list)
+        if consistent:
+            shape_list[0]['S'] = (0, len(shape_list))
+            shape_list = [shape_list[0]]
+        else:
+            for idx, lst in enumerate(shape_list):
+                lst['S'] = (idx, idx+1)
+        return shape_list
+
+    @staticmethod
+    def _read_dims_to_ranges(lif: LifFile, read_dims: Optional[Dict[str, int]] = None):
+        """
+        Convert the provided read_dims and file structure into ranges to iterate over
+
+        Parameters
+        ----------
+        lif: LifFile
+            The LifFile to get the ranges from
+        read_dims: Dict[str: int]
+            The list of locked dimensions
+
+        Returns
+        -------
+        Dict[Dimension: range]
+            These ranges can then be used to iterate through the specified YX images
+
+        """
+        if read_dims is None:
+            read_dims = {}
+
+        data_shape = LifReader._dims_shape(lif=lif)
+
+        if Dimensions.Scene in read_dims:
+            s_range = range(read_dims[Dimensions.Scene], read_dims[Dimensions.Scene] + 1)
+            s_dict = data_shape[s_range[0]]
+        else:
+            s_range = range(*data_shape[0][Dimensions.Scene])
+            s_dict = data_shape[0]
+
+        ans = {Dimensions.Scene: s_range}
+        for key in [Dimensions.Time, Dimensions.Channel, Dimensions.SpatialZ]:
+            if key in read_dims.keys():
+                ans[key] = range(read_dims[key], read_dims[key]+1)
+            else:
+                ans[key] = range(*s_dict[key])
+
+        return ans
```

Comment:
Can you change these variable names. `dim` instead of `key`. I don't remember what `ans` stands for.